pkg/api/http_server.go

```diff
@@ -84,84 +83,97 @@ func (hs *HTTPServer) Init() error {
 	hs.streamManager.Run(ctx)
 
-	listenAddr := fmt.Sprintf("%s:%s", setting.HttpAddr, setting.HttpPort)
-	listener, err := net.Listen("tcp", listenAddr)
-	if err != nil {
-		return errutil.Wrapf(err, "failed to open listener on address %s", listenAddr)
+	hs.httpSrv = &http.Server{
+		Addr:    fmt.Sprintf("%s:%s", setting.HttpAddr, setting.HttpPort),
+		Handler: hs.macaron,
 	}
-
-	hs.log.Info("HTTP Server Listen", "address", listener.Addr().String(), "protocol", setting.Protocol, "subUrl", setting.AppSubUrl, "socket", setting.SocketPath)
-
-	hs.httpSrv = &http.Server{Addr: listenAddr, Handler: hs.macaron}
-
-	// handle http shutdown on server context done
-	go func() {
-		<-ctx.Done()
-		// Hacky fix for race condition between ListenAndServe and Shutdown
-		time.Sleep(time.Millisecond * 100)
-		if err := hs.httpSrv.Shutdown(context.Background()); err != nil {
-			hs.log.Error("Failed to shutdown server", "error", err)
-		}
-	}()
-
 	switch setting.Protocol {
-	case setting.HTTP:
-		err = hs.httpSrv.Serve(listener)
-		if err == http.ErrServerClosed {
-			hs.log.Debug("server was shutdown gracefully")
-			return nil
-		}
 	case setting.HTTP2:
-		err = hs.listenAndServeH2TLS(listener, setting.CertFile, setting.KeyFile)
-		if err == http.ErrServerClosed {
-			hs.log.Debug("server was shutdown gracefully")
-			return nil
+		if err := hs.configureHttp2(); err != nil {
+			return err
 		}
 	case setting.HTTPS:
-		err = hs.listenAndServeTLS(listener, setting.CertFile, setting.KeyFile)
-		if err == http.ErrServerClosed {
-			hs.log.Debug("server was shutdown gracefully")
-			return nil
+		if err := hs.configureHttps(); err != nil {
+			return err
+		}
+	}
+
+	var listener net.Listener
+	switch setting.Protocol {
+	case setting.HTTP, setting.HTTPS, setting.HTTP2:
+		var err error
+		listener, err = net.Listen("tcp", hs.httpSrv.Addr)
+		if err != nil {
+			return errutil.Wrapf(err, "failed to open listener on address %s", hs.httpSrv.Addr)
 		}
 	case setting.SOCKET:
-		ln, err := net.ListenUnix("unix", &net.UnixAddr{Name: setting.SocketPath, Net: "unix"})
+		var err error
+		listener, err = net.ListenUnix("unix", &net.UnixAddr{Name: setting.SocketPath, Net: "unix"})
 		if err != nil {
-			hs.log.Debug("server was shutdown gracefully", "err", err)
-			return nil
+			return errutil.Wrapf(err, "failed to open listener for socket %s", setting.SocketPath)
 		}
 
 		// Make socket writable by group
 		if err := os.Chmod(setting.SocketPath, 0660); err != nil {
-			hs.log.Debug("server was shutdown gracefully", "err", err)
-			return nil
+			return errutil.Wrapf(err, "failed to change socket permissions")
 		}
+	default:
+		hs.log.Error("Invalid protocol", "protocol", setting.Protocol)
+		return fmt.Errorf("invalid protocol %q", setting.Protocol)
+	}
 
-		err = hs.httpSrv.Serve(ln)
-		if err != nil {
-			hs.log.Debug("server was shutdown gracefully", "err", err)
-			return nil
+	hs.log.Info("HTTP Server Listen", "address", listener.Addr().String(), "protocol",
```

Comment:
Would move this one up to line 108 (`case setting.HTTP, setting.HTTPS, setting.HTTP2:`) and possible add an info log for the `case setting.SOCKET:` as well.

Reply:
@marefr It doesn't make sense to move this log message as-is to the HTTP/HTTPS/HTTP2 case at least, since it is general to all cases (for example, the "protocol" tag describes the protocol in use, and the "socket" tag will describe the Unix socket if that is configured).

Reply:
Yeah so I mean keep the old log message which was for http/https/http2 so you'll get specific log message for http* and specific log message for socket..